### PR TITLE
issue-113,  to the folder where the code is generated

### LINKED ISSUE #113: To the folder where the code is generated
Steps to reproduce: go run main.go --spec-path ~/go/src/github.com/omc-college/management-system/api/rbac/rbac-api.yaml --output-path ~/go/src/github.com/omc-college/management-system/pkg/models
Current result: the result is written to a gofile named Models and only to those folders that have already been created
Expected Result: The result is written to a gofile named Models and to any folder specified on the path. If the folder is missing, it is created and the result of code generation is written to it

### PR DESCRIPTION
**Pull request by TsybenkoMaria**
closes #113

### CODE REVIEW

File: cmd/codegen/main.go [hunk 1]
```diff
@@ -17,9 +17,13 @@ import (
 func main() {
 	var buffer = bytes.NewBufferString("")
 	var err error
+ 	var prefix, sufix string
+
+	sufix = "/"
+	prefix = ".go"
```
Comment (DoodgeMatvey) [inline cmd/codegen/main.go:23]: You can immediately write the value when declaring

File: cmd/codegen/main.go [hunk 2]
```diff
@@ -34,7 +38,7 @@ func main() {
 
 	absPath, err := filepath.Abs(*outputPath)
 	if err != nil {
-		logrus.Fatalf("can not get absolute path of output directory %s, err: %s", *outputPath, err)
+		logrus.Fatalf("cannot get absolute path of output directory %s, err: %s", *outputPath, err)
```
Comment (DoodgeMatvey) [inline cmd/codegen/main.go:41]: Delete unnecessary variable and make err like err.Error()

File: cmd/codegen/main.go [hunk 3]
```diff
@@ -44,26 +48,44 @@ func main() {
...
 	formatedTypesSource, err := format.Source(buffer.Bytes())
 	if err != nil {
-		logrus.Fatalf("can not  format source types %s, err:", err)
+		logrus.Fatalf("cannot  format source types: %s", err.Error())
 	}
 
-	newTypesFile, err := os.Create(*outputPath)
+	if dir, err := os.Stat(*outputPath); os.IsNotExist(err) {
+		os.MkdirAll(*outputPath, 0700)
+		print(dir)
```
Comment (taaraora) [inline cmd/codegen/main.go:72]: remove this print please

File: cmd/codegen/main.go [hunk 4]
```diff
@@ -44,26 +48,44 @@ func main() {
...
+		panic(err)
+		logrus.Fatalf("cannot read file: %s", err.Error())
 	}
 
 	loader := openapi3.NewSwaggerLoader()
 
 	doc, err := loader.LoadSwaggerFromData(specBytes)
 	if err != nil {
-		logrus.Fatalf("can not parse OpenApi spec %s, err:", specBytes, err)
+		panic(err)
```
Comment (taaraora) [inline cmd/codegen/main.go:59]: this panic is not needed.

File: cmd/codegen/main.go [hunk 5]
```diff
@@ -44,26 +48,44 @@ func main() {
 
 	specBytes, err := ioutil.ReadFile(*specPath)
 	if err != nil {
-		logrus.Fatalf("can not read file %s, err:", *specPath, err)
+		panic(err)
```
Comment (taaraora) [inline cmd/codegen/main.go:51]: this panic is not needed.

File: cmd/codegen/main.go [hunk 6]
```diff
@@ -34,7 +37,8 @@ func main() {
 
 	absPath, err := filepath.Abs(*outputPath)
 	if err != nil {
-		logrus.Fatalf("can not get absolute path of output directory %s, err: %s", *outputPath, err)
+		panic(err)
```
Comment (taaraora) [inline cmd/codegen/main.go:40]: this panic is not needed.

File: cmd/codegen/main.go [hunk 7]
```diff
@@ -44,26 +48,44 @@ func main() {
...
 
-	newTypesFile, err := os.Create(*outputPath)
+	if dir, err := os.Stat(*outputPath); os.IsNotExist(err) {
+		os.MkdirAll(*outputPath, 0700)
+		print(dir)
+	}
+
+	fileName := filepath.Base(*specPath)
+	if err != nil {
+		panic(err)
```
Comment (taaraora) [inline cmd/codegen/main.go:77]: this panic is not needed.

File: cmd/codegen/main.go [hunk 8]
```diff
@@ -44,26 +48,44 @@ func main() {
...
+	}
+
+	outputName := strings.Split (strings.TrimSuffix(fileName, filepath.Ext(fileName)), "-")
+
+	outputFileName := suffix + outputName[0] + prefix
+
+	newTypesFile, err := os.Create(*outputPath + outputFileName)
 	if err != nil {
-		logrus.Fatalf("can not create the named file %s, err:", *outputPath, err)
+		panic(err)
```
Comment (taaraora) [inline cmd/codegen/main.go:87]: this panic is not needed.

File: pkg/codegen/models/ims.go [hunk 9]
```diff
@@ -0,0 +1,41 @@
+package codegen
+
+type ChangePasswordRequest struct {
+	ExistingPassword string `json:"existingpassword"`
```
Comment (taaraora) [inline pkg/codegen/models/ims.go:4]: JSON tags should be CamelCased

File: pkg/codegen/models/rbac.go [hunk 10]
```diff
@@ -0,0 +1,30 @@
+package codegen
```
Comment (taaraora) [inline pkg/codegen/models/rbac.go:1]: the package should be equal to the name of the folder where file is generated

File: cmd/codegen/main.go [hunk 11]
```diff
@@ -44,26 +48,44 @@ func main() {
...
 	generateNewTypes(doc, buffer)
 
 	formatedTypesSource, err := format.Source(buffer.Bytes())
 	if err != nil {
-		logrus.Fatalf("can not  format source types %s, err:", err)
+		logrus.Fatalf("cannot  format source types: %s", err.Error())
 	}
 
-	newTypesFile, err := os.Create(*outputPath)
+	if dir, err := os.Stat(*outputPath); os.IsNotExist(err) {
```
Comment (taaraora) [inline cmd/codegen/main.go:70]: please handle this error properly

**Pull request merged**